mqtt/mqtt-bridge/src/persist/waking_state/mod.rs

```diff
@@ -21,35 +20,142 @@ pub mod memory;
 /// Thus all implementations will share the below test suite.
 #[async_trait]
 pub trait StreamWakeableState {
-    fn insert(&mut self, key: Key, value: Publication) -> Result<(), PersistError>;
+    async fn insert(&self, key: Key, value: Publication) -> Result<(), PersistError>;
```

Comment:
This is something we should probably chat about as a team. 

Having async_trait will create an extra level of indirection (it desugars into Box<dyn Future...>). If we are settling on a single implementation for storage, it might be worth to stick to concrete future types kept on stack.